tests/main/snap-confine-undesired-mode-group/task.yaml

```diff
@@ -0,0 +1,48 @@
+summary: the snap-{run,confine,exec} chain does not create files with undesired properties.
+prepare: |
+    # Install a snap with opengl and joystick plugs.
+    # This gives us all of the usual snap-confine configuration, along with all
+    # the cgroups that we create. 
+    snap pack test-snapd-app
+    snap install --dangerous ./test-snapd-app_1.0_all.snap
+    snap connect test-snapd-app:opengl
+    snap connect test-snapd-app:joystick
+execute: |
+    # Run the snap as a non-root user.
+    su test -c 'snap run test-snapd-app.sh -c /bin/true'
+
+    # Look for files that are owned by the test user, group owned by the test
+    # user or are world-writable in /run/snapd as well as in /sys/fs/cgroup
+    # trees.
+    for dname in /run/snapd /sys/fs/cgroup; do
+        find "$dname" -user test >> wrong-user.txt
```

Comment:
Looking at the log from the failed test, perhaps you could use `find ... -ls -user test >> wrong-user.txt` and similarly in the other `find` invocations.

Reply:
> BTW, the test is failing:

Yes, master is buggy. I need a moment of Jamie's time to discuss my approach towards fixing this. There is disagreement about what to do.